Evaluating Genetic Regulators of MicroRNAs Using Machine Learning Models
This project leverages machine learning to uncover regulatory relationships between genes and microRNAs (miRNAs) by integrating RNA-seq and miRNA-seq data from solid tumor and healthy samples.

Workflow Overview
Data Download

Bulk RNA-seq and miRNA-seq data for 10,000 patients were retrieved from the GDC Portal via APIs.

Detailed instructions for this process can be found in the download section of gdcportal.

Data Integration

A manifest was generated using the TCGAutils library from Bioconductor in R to accurately align RNA-seq data with each patient's miRNA-seq data.

File: manifest_integration_rnaseq_mirnaseq_uuid.R

Data Extraction, Transformation, and Loading (ETL)

Using the manifest, the RNA-seq and miRNA-seq data were extracted, transformed, and loaded into a format suitable for machine learning.

File: extract_mirna_rnaseq3.py

Machine Learning Modeling

Models were trained with Scikit-learn to predict the state of miRNAs from patientsâ€™ gene expression levels, thereby uncovering miRNA-gene regulatory relationships.

File: ridge_ensemble3.py

miRNA Annotation and Enrichment Analysis

The top predictive features from the models were used to annotate miRNAs based on their molecular functions and disease enrichments through Over-Representation Analysis.

File: mirna_disease_enrichment_analysis1.py

Citation
If you use or reference this work, please cite as follows:

Cihan M, Anyaegbunam UA, Albrecht S, Andrade-Navarro MA, Sprang M. Evaluating Genetic Regulators of MicroRNAs Using Machine Learning Models. BioRxiv 2025 doi:10.1101/2025.01.09.632215. Under Review in Cell Patterns.
